javascript/ql/src/semmle/javascript/security/dataflow/CleartextLoggingCustomizations.qll

```diff
@@ -32,6 +32,22 @@ module CleartextLogging {
    * A barrier for clear-text logging of sensitive information.
    */
   abstract class Barrier extends DataFlow::Node { }
+  
+  /**
+   * A call to `.replace()` that seems to mask 
+   */
+  class MaskingReplacer extends Barrier, DataFlow::MethodCallNode {
+  	MaskingReplacer() {
+  	  this.getCalleeName() = "replace" and
+  	  exists(RegExpLiteral reg| 
+  	    reg = this.getArgument(0).getALocalSource().asExpr() and
+  	    reg.getFlags().regexpMatch("(?i).*g.*") and
+  	    reg.getRoot().getRawValue().regexpMatch(".*\\..*")
```

Comment:
This will match any regular expression with a dot in, right? Even `/\./` and `/[.]/`.
Can you check if `reg.getRoot()` contains/is `RegExp::RegExpDot` instead?

Ideally, we would also support the unlikely `new RegExp(".", "g")`, but that can wait until we have unified the treatment of the RegExp constructor and RegExpLiteral elsewhere.